From 758ad699cfad8fb5151f336189b78a208fcaa850 Mon Sep 17 00:00:00 2001
From: Michal Marek <mmarek@suse.cz>
Date: Mon, 11 Mar 2013 13:45:08 +0100
Subject: [PATCH] MODSIGN: Support for signing_key.{x509,priv} in the M=
 directory
Patch-mainline: Submitted 2013-03-11

Allow for the module signing keypair to be supplied when building an
external module.

Signed-off-by: Michal Marek <mmarek@suse.cz>
---
 Makefile |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index a05ea42..a0c5436 100644
--- a/Makefile
+++ b/Makefile
@@ -721,8 +721,13 @@ export mod_strip_cmd
 
 
 ifdef CONFIG_MODULE_SIG_ALL
+ifeq ($(KBUILD_EXTMOD),)
 MODSECKEY = ./signing_key.priv
 MODPUBKEY = ./signing_key.x509
+else
+MODSECKEY = $(firstword $(wildcard $(KBUILD_EXTMOD)/signing_key.priv) ./signing_key.priv)
+MODPUBKEY = $(firstword $(wildcard $(KBUILD_EXTMOD)/signing_key.x509) ./signing_key.x509)
+endif
 export MODPUBKEY
 mod_sign_cmd = perl $(srctree)/scripts/sign-file $(CONFIG_MODULE_SIG_HASH) $(MODSECKEY) $(MODPUBKEY)
 else
-- 
1.7.10.4

